### PR TITLE
add support for LoginRequiredMiddleware with login_not_required decor…

### PR DESCRIPTION
**Pull request by dulmandakh**
#### Description

This PR adds support for LoginRequiredMiddleware introduced in Django 5.1, or views provided by this package don't require login.

#### Checklist:

- [x] I have added the relevant tests for this change.
- [x] I have added an item to the Pending section of ``docs/changes.rst``.

### CODE REVIEW

File: tests/urls.py [hunk 1]
```diff
@@ -26,5 +26,5 @@
     path("ajax/", views.ajax_view),
     path("login_without_redirect/", LoginView.as_view(redirect_field_name=None)),
     path("admin/", admin.site.urls),
-    path("__debug__/", include("debug_toolbar.urls")),
+    path("__debug__/", include("debug_toolbar.urls", namespace="djdt")),
```
Comment (matthiask) [inline tests/urls.py:29]: What's the reason for this change?

File: debug_toolbar/decorators.py [hunk 2]
```diff
@@ -5,9 +5,17 @@
...
+    from django.contrib.auth.decorators import login_not_required
+except ImportError:
+
+    def login_not_required(func):
+        return func
+
 
 def require_show_toolbar(view):
     @functools.wraps(view)
+    @login_not_required
```
Comment (matthiask) [inline debug_toolbar/decorators.py:18]: I don't think `require_show_toolbar` necessarily implies `login_not_required`. In practice it does, but I'd prefer if the decorator was added to the view function themselves, even though you have to scatter the `@login_not_required` invocations over the code base. 

I have added `authlib/_compat.py` (with a leading underscore to signify that this is purely internal and shouldn't be relied on), the same thing may make sense here.

File: docs/changes.rst [hunk 3]
```diff
@@ -11,6 +11,7 @@ Pending
   `Google Summer of Code Project 2024
   <https://summerofcode.withgoogle.com/programs/2024/projects/iXVvyGYp>`__.
 * Added Django 5.1 to the CI matrix.
+* Added support for LoginRequiredMiddleware introduced in Django 5.1.
```
Comment (matthiask) [inline docs/changes.rst:14]:
```suggestion
* Added support for the ``LoginRequiredMiddleware`` introduced in Django 5.1.
```

File: tests/test_login_not_required.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    def test_login_not_required(self):
+        self.assertEqual(self.client.get(reverse("djdt:render_panel")).status_code, 200)
+
+
+# DEBUG_TOOLBAR_CONFIG = {
+#     # Django's test client sets wsgi.multiprocess to True inappropriately
+#     "RENDER_PANELS": False,
+#     # IS_RUNNING_TESTS must be False even though we're running tests because we're running the toolbar's own tests.
+#     "IS_RUNNING_TESTS": False,
+# }
```
Comment (matthiask) [inline tests/test_login_not_required.py:31]:
```suggestion
```

**Review by matthiask — COMMENTED** [review]
Thanks for working on this!

Here's an example of an app adding support for `login_not_required` while actually testing the behavior:
https://github.com/matthiask/django-authlib/commit/2239a1c1d313e1f5a8db14531d0458fbc19f5c6e

File: tests/urls.py (hunk 1 above)
Comment (dulmandakh) [reply tests/urls.py:29]: just to use reverse in tests

**Comment by dulmandakh** [comment]
> Thanks for working on this!
> 
> Here's an example of an app adding support for `login_not_required` while actually testing the behavior: [matthiask/django-authlib@2239a1c](https://github.com/matthiask/django-authlib/commit/2239a1c1d313e1f5a8db14531d0458fbc19f5c6e)

I contributed same support for Django OAuth Toolkit in https://github.com/jazzband/django-oauth-toolkit/pull/1454. Thanks for the reference.

File: debug_toolbar/decorators.py (hunk 2 above)
Comment (dulmandakh) [reply debug_toolbar/decorators.py]: changed the implementation as suggested

File: tests/test_login_not_required.py (hunk 4 above)
Comment (dulmandakh) [reply tests/test_login_not_required.py]: removed comments

**Comment by dulmandakh** [comment]
@matthiask please review and merge if this PR looks good

**Comment by matthiask** [comment]
Thanks!
